src/neo4j/_async_compat/network/_bolt_socket.py

```diff
@@ -548,6 +551,12 @@ def _connect(cls, resolved_address, timeout, keep_alive):
                 s = socket(AF_INET6)
             else:
                 raise ValueError(f"Unsupported address {resolved_address!r}")
+            try:
```

Comment:
If the exception is raised does that surface to the user or get logged and the driver continues to operate using Nagels algorithm?